src/components/StatusUpdater/index.tsx

```diff
@@ -7,25 +7,43 @@ import { UserInfo } from '@respond/types/userInfo';
 import { SplitButton } from '../SplitButton';
 import { useFormLogic, UpdateStatusForm } from './UpdateStatusForm';
 
-const options = [
-  { id: ResponderStatus.Unavailable, text: 'Not Available' },
-  { id: ResponderStatus.Standby, text: 'Stand By' },
-  { id: ResponderStatus.SignedIn, text: 'Sign In' },
-  { id: ResponderStatus.SignedOut, text: 'Sign Out' },
-]
-const optionTexts = options.reduce((accum, cur) => ({ ...accum, [cur.id]: cur.text }), {} as Record<string, string>);
+const options = {
+  standby: { id: ResponderStatus.Standby, text: 'Stand By' },
+  signin: { id: ResponderStatus.SignedIn, text: 'Sign In' },
+  signout: { id: ResponderStatus.SignedOut, text: 'Sign Out' },
+}
+const optionTexts = Object.values(options).reduce((accum, cur) => ({ ...accum, [cur.id]: cur.text }), {} as Record<string, string>);
+
+function isFuture(time: number) {
+  return (time - 60 * 60 * 1000) > new Date().getTime();
+};
 
 function getRecommendedAction(current: ResponderStatus|undefined, startTime: number): ResponderStatus {
-  const now = new Date().getTime();
   if (current === ResponderStatus.SignedIn) {
     return ResponderStatus.SignedOut;
   }
-  if (startTime - 60 * 60 * 1000 > now) {
-    return ResponderStatus.Standby;
+  if (isFuture(startTime)) {
+    return current === ResponderStatus.Standby ? ResponderStatus.SignedOut : ResponderStatus.Standby;
   }
   return ResponderStatus.SignedIn;
 }
 
+function getCurrentOptions(current: ResponderStatus|undefined, startTime: number) {
+  if (isFuture(startTime)) {
+    return current === ResponderStatus.Standby ? [options.signout] : [options.standby];
+  }
+  if (current === ResponderStatus.Standby) {
+    return [options.signin, options.signout];
+  }
+  if (current === ResponderStatus.SignedIn) {
+    return [options.signout];
+  }
+  if (current === ResponderStatus.SignedOut) {
+    return [options.signin, options.standby];
+  }
+  return [options.signin, options.standby];

```

Comment:
Could the non-future start time transitions be described with a const map of some kind? Another more readable option may be to use a switch statement.